From a3f66b1de7df4274d10bd48d1f14c1be49082879 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Tue, 27 Oct 2026 13:20:18 +0100
Subject: [PATCH 50/N] Cygwin: cygcheck: preserve constness when scanning
 extensions

The extension scan in `find_on_path()` fails to compile with GCC v16 because it
tries to assign a `const char *` return value to a `char *` variable.

Heed the pathname's read-only contract rather than casting away `const`. The
suffix is only inspected, therefore nothing in the extension detection and
path-search logic needs to change.

Assisted-by: GPT-6.1 Sol
Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/utils/mingw/cygcheck.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/winsup/utils/mingw/cygcheck.cc b/winsup/utils/mingw/cygcheck.cc
index 22c7d18..b5994e7 100644
--- a/winsup/utils/mingw/cygcheck.cc
+++ b/winsup/utils/mingw/cygcheck.cc
@@ -443,7 +443,7 @@ find_on_path (const char *in_file, const char *ext, bool showall = false,
       return 0;
     }
 
-  char *hasext = strrchr (file, '.');
+  const char *hasext = strrchr (file, '.');
   if (hasext && !strpbrk (hasext, "/\\"))
     ext = "";
 
